### PR TITLE
* : return error if get progress status failed

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### What problem does this PR solve?

Issue Number: Ref #4640.

### What is changed and how does it work?
This PR will return err if the unexpected situation happens, also change back the tombstone condition from size to count.

```commit-message
```

### Check List

Tests

- Unit test
### Release note

```release-note
None
```

### CODE REVIEW

**Review by JmPotato — COMMENTED** [review]
Do we need to define these errors in `pkg/errs`?

**Review by AndreMouche — COMMENTED** [review]
Could you please add some tests for the failed cases?

**Comment by rleungx** [comment]
> Could you please add some tests for the failed cases?

We do have a test in https://github.com/tikv/pd/blob/6554d63c7350f9bcbe2f5ff50d81808398aade67/tests/server/api/api_test.go#L559-L563

File: pkg/progress/progress.go [hunk 1]
```diff
@@ -131,12 +133,16 @@ func (m *Manager) GetProgresses(filter func(p string) bool) []string {
...
-func (m *Manager) Status(progress string) (process, leftSeconds, currentSpeed float64) {
+func (m *Manager) Status(progress string) (process, leftSeconds, currentSpeed float64, err error) {
 	m.RLock()
 	defer m.RUnlock()
 
 	if p, exist := m.progesses[progress]; exist {
 		process = 1 - p.remaining/p.total
+		if process < 0 {
+			err = errs.ErrProgressWrongStatus.FastGenByArgs(fmt.Sprintf("the remaining: %v is larger than the total: %v", p.remaining, p.total))
+			return
```
Comment (AndreMouche) [inline pkg/progress/progress.go:144]: I mean test to cover L143-L144 😄

**Comment by rleungx** [comment]
FYI, made more changes @JmPotato @AndreMouche

**Review by AndreMouche — APPROVED** [review]
LGTM

**Review by JmPotato — APPROVED** [review]

**Pull request merged**